pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -0,0 +1,485 @@
+        checkArgument(!isNullOrEmpty(config.getString(SECRET_KEY)));
+        String accessKey = config.getString(ACCESS_KEY);
+        String secretKey = config.getString(SECRET_KEY);
+        try {
+            AwsBasicCredentials awsBasicCredentials = AwsBasicCredentials.create(accessKey, secretKey);
+            s3Client = S3Client.builder()
+                    .region(Region.AP_SOUTHEAST_1)
+                    .credentialsProvider(StaticCredentialsProvider.create(awsBasicCredentials))
+                    .build();
+
+        } catch (S3Exception e) {
+            throw new RuntimeException("Could not initialize S3PinotFS", e);
+        }
+    }
+
+    private HeadObjectResponse getS3ObjectMetadata(URI uri) throws IOException {
+            URI base = getBase(uri);
+            String path = sanitizePath(base.relativize(uri).getPath());
+            HeadObjectRequest headObjectRequest = HeadObjectRequest.builder()
+                    .bucket(uri.getHost())
+                    .key(path)
+                    .build();
+
+            return s3Client.headObject(headObjectRequest);
+    }
+
+    private boolean isPathTerminatedByDelimiter(URI uri) {
+        return uri.getPath().endsWith(DELIMITER);
+    }
+
+    private String normalizeToDirectoryPrefix(URI uri) throws IOException {
+        requireNonNull(uri, "uri is null");
+        URI strippedUri = getBase(uri).relativize(uri);
+        if (isPathTerminatedByDelimiter(strippedUri)) {
+            return sanitizePath(strippedUri.getPath());
+        }
+        return sanitizePath(strippedUri.getPath() + DELIMITER);
+    }
+
+    private URI normalizeToDirectoryUri(URI uri) throws IOException {
+        if (isPathTerminatedByDelimiter(uri)) {
+            return uri;
+        }
+        try {
+            return new URI(uri.getScheme(), uri.getHost(), sanitizePath(uri.getPath() + DELIMITER), null);
+        } catch (URISyntaxException e) {
+            throw new IOException(e);
+        }
+    }
+
+    private String sanitizePath(String path) {
+        path = path.replaceAll(DELIMITER + "+", DELIMITER);
+        if (path.startsWith(DELIMITER) && !path.equals(DELIMITER)) {
+            path = path.substring(1);
+        }
+        return path;
+    }
+
+    private URI getBase(URI uri) throws IOException {
+        try {
+            return new URI(uri.getScheme(), uri.getHost(), null, null);
+        } catch (URISyntaxException e) {
+            throw new IOException(e);
+        }
+    }
+
+    private boolean existsFile(URI uri) throws IOException {
+        try {
+            URI base = getBase(uri);
+            String path = sanitizePath(base.relativize(uri).getPath());
+            HeadObjectRequest headObjectRequest = HeadObjectRequest.builder()
+                    .bucket(uri.getHost())
+                    .key(path)
+                    .build();
+
+            s3Client.headObject(headObjectRequest);
+            return true;
+        }catch(NoSuchKeyException e){
```

Comment:
Missing space between `}` and `catch`. Please format the files with codestyle config in https://github.com/apache/incubator-pinot/tree/master/config